File: src/main/java/com/arnas/klatrebackend/features/routes/RouteService.java
package com.arnas.klatrebackend.features.routes;

import org.springframework.lang.NonNull;

public interface RouteService {

    long addRoute(@NonNull RouteDTO routeDTO, long userId);
    void updateRoute(@NonNull RouteUpdateDTO routeDTO, long userId);
    void deleteRoute(long routeId);
    @NonNull
    RouteResponse getRoutesByPlace(long placeId, int page, int limit);

}

File: src/main/kotlin/com/arnas/klatrebackend/features/routes/RouteService.kt
package com.arnas.klatrebackend.features.routes

interface RouteService {
    fun addRoute(routeDTO: RouteDTO, userId: Long): Long
    fun updateRoute(routeDTO: RouteUpdateDTO, userId: Long)
    fun deleteRoute(routeId: Long)
    fun getRoutesByPlace(placeId: Long, page: Int, limit: Int): RouteResponse
}

